context/app/static/js/components/search/Facets/DateRangeFacet.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useCallback, useState, useEffect } from 'react';
+import Stack from '@mui/material/Stack';
+import { DatePicker, DatePickerProps } from '@mui/x-date-pickers/DatePicker';
+import { DateValidationError } from '@mui/x-date-pickers/models';
+import { format } from 'date-fns/format';
+
+import { useSearch } from '../Search';
+import { isDateFilter, useSearchStore } from '../store';
+import { getFieldLabel } from '../fieldConfigurations';
+import FacetAccordion from './FacetAccordion';
+
+interface DateRangeFacetProps {
+  field: string;
+}
+
+function DatePickerComponent({
+  minDate,
+  maxDate,
+  ...rest
+}: DatePickerProps<Date> & Required<Pick<DatePickerProps<Date>, 'minDate' | 'maxDate'>>) {
+  const [error, setError] = React.useState<DateValidationError | null>(null);
+
+  const errorMessage = React.useMemo(() => {
+    switch (error) {
+      case 'maxDate':
+        return `Please select a date of ${format(maxDate, 'MMMM yyyy')} or before.`;
+
+      case 'minDate': {
+        return `Please select a date of ${format(minDate, 'MMMM yyyy')} or after.`;
+      }
+
+      case 'invalidDate': {
+        return 'The date is not valid.';
+      }
+
+      default: {
+        return '';
+      }
+    }
+  }, [error, minDate, maxDate]);
+
+  return (
+    <DatePicker
+      sx={(theme) => ({
+        '.MuiOutlinedInput-input, .MuiInputLabel-root, .MuiMonthCalendar-root': {
+          fontSize: theme.typography.subtitle2.fontSize,
+        },
+      })}
+      views={['month', 'year']}
+      onError={(newError) => setError(newError)}
+      minDate={minDate}
+      maxDate={maxDate}
+      slotProps={{
+        textField: {
+          helperText: errorMessage,
+        },
+      }}
+      {...rest}
+    />
+  );
+}
+
+function DateRangeFacet({
+  field,
+  min,
+  max,
+  aggMin,
+  aggMax,
+}: DateRangeFacetProps & { min: number; max: number; aggMin: number; aggMax: number }) {
+  const filterDate = useSearchStore((state) => state.filterDate);
+  const [values, setValues] = useState([min, max]);
+
+  // Reset slider position when filter chip is deleted.
+  useEffect(() => {
+    setValues([min, max]);
+  }, [min, max, setValues]);
+
+  const filterMin = useCallback(
+    (value: Date | null) => {
+      if (value) {
```

Comment:
Under which circumstances is this null? Should `filterDate` run to unset `min` if the value is null? It looks like the values are initially `undefined` in `buildFacets`, so perhaps resetting to that would make sense?